src/server/detail/snapshot_storage.cc

```diff
@@ -116,14 +120,16 @@ io::Result<std::string, GenericError> FileSnapshotStorage::LoadPath(std::string_
   if (fs::exists(fl_path))
     return fl_path.generic_string();
 
-  SubstituteFilenameTsPlaceholder(&fl_path, "*");
+  SubstituteFilenamePlaceholders(&fl_path, "*", "*", "*", "*");
   if (!fl_path.has_extension()) {
     fl_path += "*";
   }
   io::Result<io::StatShortVec> short_vec = io::StatFiles(fl_path.generic_string());
   if (short_vec) {
-    // io::StatFiles returns a list of sorted files. Because our timestamp format has the same
-    // time order and lexicographic order we iterate from the end to find the latest snapshot.
+    std::sort(short_vec->begin(), short_vec->end(),
+              [](const io::StatShort& l, const io::StatShort& r) {
+                return std::difftime(l.last_modified, r.last_modified) < 0;
```

Comment:
I'm not sure if this is the way to go.
Copying files, or touching them either manually or with some tool, may affect last modified. I think we should look at the actual date format, as that's what the users would assume will happen.
Granted, it's harder (and sometimes even unclear I guess - if timestamp is not present for example), but I think deferring to last modified should be a last resort

Reply:
If a user uses `{Y}`, `{m}`, `{d}` notation, it's not guaranteed that dictionary order is aligned with the recency.
I think for most cases using sort using `mtime` should work well: we do not have to assume that it's our responsibility to handle unexpected metadata changes.

Reply:
I agree that it's harder. I just see some nasty data loss cases if we use mdata. But sure, if you think it's ok then I withhold my comment.

Just thinking out loud though, if timestamp is present we can try to detect where it is. Sure, the format might have some such characters, but we can always revert to mdata.
Your call. Feel free to ignore.

Reply:
@chakaz I like your idea more, but for now, I've decided to leave as is. Honestly, I would prefer to remove {timestamp} macro at all and provide full set of timestamp macros.

Reply:
Sure thing, your call :)